integration/src/test/java/org/ldaptive/ssl/TLSSocketFactoryTest.java

```diff
@@ -167,21 +231,162 @@ public void connectTLS(final String url)
   public void connectSSL(final String url)
     throws Exception
   {
-    // with no trusted certificates, connection should fail
-    final ConnectionConfig cc = createSSLConnectionConfig(url);
```

Comment:
You might consider breaking out the tests in this class into individual test methods so you can drive test failure with annotations:

    @Test(expectedAnnotations = {A.class, B.class, C.class}

This would also address the problem of different providers throwing different exception types on SSL handshake errors.